### PR TITLE
[7X] Suppress compiler warning.

### PR DESCRIPTION
**Pull request by higuoxing**
optimize_query is qualified by static and defined in orca.h. If build gpdb w/o ORCA and include orca.h without using optimize_query, there will be warnings.

```
In file included from explain.c:56:
../../../src/include/optimizer/orca.h:31:1: warning: 'optimize_query' defined but not used [-Wunused-function]
   31 | optimize_query(Query *parse, int cursorOptions, ParamListInfo boundParams)
      | ^~~~~~~~~~~~~~
```

### CODE REVIEW

File: src/backend/optimizer/plan/planner.c [hunk 1]
```diff
@@ -370,7 +370,11 @@ standard_planner(Query *parse, int cursorOptions, ParamListInfo boundParams)
 		if (gp_log_optimization_time)
 			INSTR_TIME_SET_CURRENT(starttime);
 
+#ifdef USE_ORCA
```
Comment (kainwen) [inline src/backend/optimizer/plan/planner.c:373]: Add a comment `silent compiler`.

**Review by kainwen — APPROVED** [review]

**Review by zhaorui20005 — APPROVED** [review]

File: src/backend/optimizer/plan/planner.c [hunk 2]
```diff
@@ -370,7 +370,11 @@ standard_planner(Query *parse, int cursorOptions, ParamListInfo boundParams)
...
 		result = optimize_query(parse, cursorOptions, boundParams);
+#else
+		result = NULL;
```
Comment (beeender) [inline src/backend/optimizer/plan/planner.c:376]: Shall we do `Assert(false)` here just like before?

Comment (zhaorui20005) [reply src/backend/optimizer/plan/planner.c:376]: What is the functionality of the `Assert(false)`?

Comment (beeender) [reply src/backend/optimizer/plan/planner.c:376]: > What is the functionality of the Assert(false)?
Just panic in the debug build. (do nothing in *gpdb* release build)

Comment (zhaorui20005) [reply src/backend/optimizer/plan/planner.c:376]: If configured with --disable-orca, the code inside this block (between line 364 and 390)will be optimized away.

**Review by gpopt — APPROVED** [review]

**Pull request merged**